### ISSUE TITLE
[quoted-strings] False positive on strings with a colon

### ISSUE THREAD

**Issue by Zamiell** [adrienverge/yamllint#799]
Hello and thanks for the excellent linter. This is a bug report.

The following YAML will fail the "quoted-strings" rule:

```yaml
- text: "foo:"
```

However, the quotes are necessary, because the colon will make it have a syntax error:

```
 error    syntax error: mapping values are not allowed here (syntax)
```
